socorro/signature/generator.py

```diff
@@ -72,15 +73,54 @@ def to_dict(self):
 
 
 class SignatureGenerator:
-    def __init__(self, pipeline=None, error_handler=None):
+    def __init__(self, ruleset=None, error_handler=None, **cfg):
         """
-        :arg pipeline: list of rules to use for signature generation
-        :arg error_handler: error handling function with signature
+        :param ruleset: list of rule classes to use for signature generation
+        :param error_handler: error handling function with signature
             ``fun(signature_data, exc_info, extra)``
+        :param cfg: configuration keyword arguments to pass to rules being
```

Comment:
I'm not thrilled about the `cfg` kwargs thing, but it was either that or carrying configuration through the layers explicitly or setting up a configuration system ... I figure we can start with this and if/when we see it causing problems, we can adjust as necessary.